src/arithmetic/eval.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{
+    arithmetic::circuit::ArithmeticCircuit,
+    bgg::{
+        encoding::BggEncoding,
+        public_key::BggPublicKey,
+        sampler::{BGGEncodingSampler, BGGPublicKeySampler},
+    },
+    gadgets::crt::biguint_to_crt_poly,
+    lookup::{
+        poly::PolyPltEvaluator,
+        simple_eval::{SimpleBggEncodingPltEvaluator, SimpleBggPubKeyEvaluator},
+    },
+    matrix::PolyMatrix,
+    poly::{Poly, PolyParams},
+    sampler::{PolyHashSampler, PolyTrapdoorSampler, PolyUniformSampler},
+};
+use num_bigint::BigUint;
+use std::path::PathBuf;
+use tracing::info;
+
+const TAG_BGG_PUBKEY: &[u8] = b"BGG_PUBKEY";
+
+impl<P: Poly> ArithmeticCircuit<P> {
+    pub fn evaluate_with_poly(&self, params: &P::Params, inputs: &[BigUint]) -> Vec<P> {
+        let one = P::const_one(params);
+        let mut all_input_polys = Vec::new();
+
+        for input in inputs {
+            let crt_limbs = biguint_to_crt_poly(self.limb_bit_size, params, input);
+            all_input_polys.extend(crt_limbs);
+        }
+
+        let plt_evaluator = PolyPltEvaluator::new();
+        let results = self.poly_circuit.eval(params, &one, &all_input_polys, Some(plt_evaluator));
+        results
+    }
+
+    pub fn evaluate_with_bgg_pubkey<M, SH, ST, SU>(
+        &self,
+        params: &P::Params,
+        num_inputs: usize,
+        seed: [u8; 32],
+        dir_path: PathBuf,
+        d: usize,
+        b_epsilon: M,
```

Comment:
Can you rename b_epsilon and b_epsilon with pub_matrix and trapdoor, respectively?